src/Agent.Worker/RetryHelper.cs

```diff
@@ -48,6 +60,49 @@ public async Task<T> Retry<T>(Func<Task<T>> action, Func<int, int> timeDelayInte
 
         }
 
+
+        public async Task RetryStep(Func<Task> action, Func<int, int> timeDelayInterval)
+        {
+            int retryCounter = 0;
+            do
+            {
+                using (new SimpleTimer($"RetryHelper Method:{action.Method} ", Debug))
+                {
+                    try
+                    {
+                        if (retryCounter > 0)
+                        {
+                            ExecutionContext.ReInitializeForceCompleted();
+                        }
+
+                        Debug($"Invoking Method: {action.Method}. Attempt count: {retryCounter}");
+                        await action();
+
+                        if (ExecutionContext.Result != TaskResult.Failed || ExhaustedRetryCount(retryCounter))
+                        {
+                            return;
+                        }
+                        else
+                        {
+                            string exceptionMessage = $"Task result ${ExecutionContext.Result}";
+                            ExecutionContext.Result = null;
+                            Warning($"RetryHelper encountered task failure, will retry (attempt #: {retryCounter + 1}) out of {this.MaxRetries}");
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        if (!ShouldRetryStepOnException(ex) || ExhaustedRetryCount(retryCounter))
+                        {
+                            throw;
+                        }
+                        Warning($"RetryHelper encountered exception, will retry (attempt #: {retryCounter} {ex.Message}) ");
+                    }
+                    await Task.Delay(timeDelayInterval(retryCounter));
```

Comment:
```CSharp
public static Task Delay(int millisecondsDelay, CancellationToken cancellationToken);
```
Let's also pass the cancellation token to be able to interrupt tasks while we awaiting the end of timeout between retry attempts.

Also, could it make sense to display a message that at the moment the task is suspended due to timeout between retry attempts?

Reply:
Fixed - passed cancellation token from execution context, added message about delay.